src/Mutex.php

```diff
@@ -40,6 +42,9 @@ public function __construct(bool $autoRelease = true)
         if ($autoRelease) {
             $locks = &$this->locks;
             register_shutdown_function(function () use (&$locks) {
+                /**
+                 * @var string $lock
```

Comment:
Instead of typing here, can you do it in property declaration?

```php
@psalm-var list<string>
```

Reply:
Docblock for locks should be enough - it works well on psalm 3.15, but psalm 4 generates an error when variable passed by reference

Example without reference
https://psalm.dev/r/01bd094997